doorman/jwt_claims.go

```diff
@@ -0,0 +1,30 @@
+package doorman
+
+import (
+	jose "gopkg.in/square/go-jose.v2"
+	jwt "gopkg.in/square/go-jose.v2/jwt"
+)
+
+// Claims is the set of information we extract from the JWT payload.
+type Claims struct {
+	Subject  string       `json:"sub,omitempty"`
+	Audience jwt.Audience `json:"aud,omitempty"`
+	Email    string       `json:"email,omitempty"`
+	Groups   []string     `json:"groups,omitempty"`
+}
+
+// ClaimExtractor is in charge of extracting meaningful info from JWT payload.
+type ClaimExtractor interface {
+	Extract(*jwt.JSONWebToken, *jose.JSONWebKey) (*Claims, error)
+}
+
```

Comment:
Maybe add a `var defaultExtractor = &defaultClaimExtractor{}`. Then you can just reuse it.  Not sure if this makes much diff to mem/cpu benchmarks. Just a matter of style.